crates/sequencer/src/sequencer.rs

```diff
@@ -591,94 +597,151 @@ where
         }
     }
 
-    async fn submit_commitment(
+    async fn try_submit_commitment(
         &mut self,
         state_diff_threshold_reached: bool,
     ) -> anyhow::Result<()> {
         debug!("Sequencer: Checking if commitment should be submitted");
-        let inscription_queue = self.da_service.get_send_transaction_queue();
-        let min_soft_confirmations_per_commitment =
-            self.config.min_soft_confirmations_per_commitment;
 
         let commitment_info = commitment_controller::get_commitment_info(
             &self.ledger_db,
-            min_soft_confirmations_per_commitment,
+            self.config.min_soft_confirmations_per_commitment,
             state_diff_threshold_reached,
         )?;
-
         if let Some(commitment_info) = commitment_info {
-            let l2_range_to_submit = commitment_info.l2_height_range.clone();
-
-            // calculate exclusive range end
-            let range_end = BatchNumber(l2_range_to_submit.end().0 + 1); // cannnot add u64 to BatchNumber directly
-
-            let soft_confirmation_hashes = self
-                .ledger_db
-                .get_soft_batch_range(&(*l2_range_to_submit.start()..range_end))?
-                .iter()
-                .map(|sb| sb.hash)
-                .collect::<Vec<[u8; 32]>>();
-
-            let commitment = commitment_controller::get_commitment(
-                commitment_info.clone(),
-                soft_confirmation_hashes,
-            )?;
-
-            debug!(
-                "Sequencer: submitting commitment for L2 range {}-{}",
-                commitment.l2_start_block_number, commitment.l2_end_block_number
-            );
-
-            let blob = borsh::to_vec(&DaData::SequencerCommitment(commitment.clone()))
-                .map_err(|e| anyhow!(e))?;
-            let (notify, rx) = oneshot_channel();
-            let request = BlobWithNotifier { blob, notify };
-            inscription_queue
-                .send(request)
-                .map_err(|_| anyhow!("Bitcoin service already stopped!"))?;
-            let tx_id = rx
-                .await
-                .map_err(|_| anyhow!("DA service is dead!"))?
-                .map_err(|_| anyhow!("Send transaction cannot fail"))?;
+            self.submit_commitment(commitment_info, false).await?;
+        }
+        Ok(())
+    }
+
+    #[instrument(level = "trace", skip(self), err, ret)]
+    pub async fn resubmit_pending_commitments(&mut self) -> anyhow::Result<()> {
+        let pending_commitments = self.pending_commitments_l2_range.lock().await.clone();
+        for (l2_start, l2_end) in pending_commitments {
+            let commitment_info = commitment_controller::CommitmentInfo {
+                l2_height_range: l2_start..=l2_end,
+            };
+            self.submit_commitment(commitment_info, true).await?;
```

Comment:
IMO, this is what i would call blind re-submission. 

Scenario:
- Sequencer sees a commitment is due
- Submits commitment
- Sequencer is stopped while commitment TX is in DA layer
- TX in DA layer goes through while sequencer is stopped
- Sequencer is started
- Sequencer resubmits an already successful commitment tx (pays fee as well)

Last step is redundant and costly.

I would be against merging this approach completely.

The alternative is coming up with a way to let the sequencer figure out which pending blocks were included on DA vs the ones that were not and resubmitting those.

Reply:
Is there a way to query transactions by the address from DA, and the details of it, so at the restart we can ask it and configure sequencer accordingly? I understand your point and I too agree that this doesn't feel nice. But I think this problem can be generalized as atomicity problem across different components of the system. I don't know of a nice solution to this problem.

Reply:
One option is to wait for L1 block sync of the sequencer which pretty much does contain information about commitments. Another approach would be to query the Da with the TX hashes we have but that's an extra step.

Reply:
I left the comment on #537 for this possible scenario actually :D